### ISSUE TITLE
IO "writing methods" need to be able to re-construct masks to have Dimensions-like structure

### ISSUE THREAD

**Issue by AlasdairE** [Quantipy/quantipy#576]
A new method in _io.py_, called `dimensionize_meta`, would cycle through the given meta and re-construct the masks in the Dimensions style, i.e. 
```python
            ORIGINAL    DIMENSIONS
MASK NAME   "Q3"        "Q3.Q3_grid"

MASK COLUMN "Q3_1"      "Q3[{Q3_1}].Q3_grid"
```
This will need to apply the column name changes to the data column names.

**Comment by jamesrkg** [comment]
This is being handled by new `Dataset` methods for general renaming purposes, plus convenience functions for generating the `dimensionize`/`undimensionize` renaming mappers for use with those methods.

**Comment by jamesrkg** [comment]
I'm adding `dimensionizing_mapper()` and `undimensionizing_mapper()` to `core.tools.dp.io.py` to address this. The returned mapper needs to be used with the new `Dataset` method `rename_dataset`.

**Comment by jamesrkg** [comment]
`dimensionizing_mapper()` and `undimensionizing_mapper()` resolved by #582.
